### PR TITLE
#85534 Update unit tests for ConfigurationMerger

### PR DESCRIPTION
**Pull request by tabman83**

### CODE REVIEW

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs [hunk 1]
```diff
@@ -18,6 +18,43 @@ namespace CaptainHook.Tests.Director
...
     {
+        private static readonly BasicAuthenticationEntity BasicAuthenticationEntity = new BasicAuthenticationEntity("username", "secret-name");
+        private static readonly OidcAuthenticationEntity OidcAuthenticationEntity = new OidcAuthenticationEntity("captain-hook-id", "kv-secret-name", "https://blah-blah.sts.eshopworld.com", new[] { "scope1" });
+
+        private static readonly BasicAuthenticationConfig BasicAuthenticationConfig = new BasicAuthenticationConfig
+        {
+            Username = "username",
+            Password = "password"
+        };
+        private static readonly OidcAuthenticationConfig OidcAuthenticationConfig = new OidcAuthenticationConfig
```
Comment (dawidkeshop) [inline src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:29]: I would keep a blank line between these two.

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs [hunk 2]
```diff
@@ -18,6 +18,43 @@ namespace CaptainHook.Tests.Director
...
+        {
+            ClientId = "captain-hook-id",
+            ClientSecret = "secret",
+            Uri = "https://blah-blah.sts.eshopworld.com",
+            Scopes = new[] { "scope1" }
+        };
+
+        private readonly Mock<ISecretProvider> _secretProvider;
+        public static IEnumerable<object[]> Data =>
```
Comment (dawidkeshop) [inline src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:38]: Does it have to be `public`? And I would give a more descriptive name about what is this `Data` needed for. :)

File: src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ public class SubscriberConfigurationBuilder
         private string _name = "event1";
         private string _subscriberName = "captain-hook";
         private string _uri = "https://blah.blah.eshopworld.com";
-        private HttpMethod _httpMethod = HttpMethod.Post;
+        private string _httpVerb = "POST";
```
Comment (chomick) [inline src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs:15]: I think POST is the default one, maybe we should pick something that is not a default method we use?

**Review by chomick — APPROVED** [review]

File: src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs (hunk 3 above)
Comment (tabman83) [reply src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs:15]: This is just for the builder, tests usually have variations through `Theory`

File: src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs [hunk 4]
```diff
@@ -86,6 +86,20 @@ public SubscriberConfigurationBuilder WithBasicAuthentication()
             return this;
         }
 
+        public SubscriberConfigurationBuilder WithAuthentication(AuthenticationConfig authenticationConfig)
```
Comment (dawidkeshop) [inline src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs:89]: I think maybe would be good to call this new method from `WithBasicAuthentication` and `WithOidcAuthentication` to be more concise?

**Review by dawidkeshop — APPROVED** [review]
Couple cosmetic suggestions to consider.

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs (hunk 1 above)
Comment (tabman83) [reply src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:29]: Done

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs (hunk 2 above)
Comment (tabman83) [reply src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:38]: It has to be public, data is only scoped to this test

File: src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs (hunk 4 above)
Comment (tabman83) [reply src/Tests/CaptainHook.TestsInfrastructure/Builders/SubscriberConfigurationBuilder.cs:89]: Absolutely, changing that

**Pull request merged**